lldb/tools/lldb-dap/EventHelper.cpp

```diff
@@ -178,15 +182,14 @@ void SendThreadStoppedEvent(DAP &dap) {
           SendThreadExitedEvent(dap, tid);
       }
     } else {
-      if (dap.log)
-        *dap.log << "error: SendThreadStoppedEvent() when process"
-                    " isn't stopped ("
-                 << lldb::SBDebugger::StateAsCString(state) << ')' << std::endl;
+      LLDB_LOG(GetLog(DAPLog::Protocol),
+               "error: SendThreadStoppedEvent() when process"
+               " isn't stopped ({0})",
+               lldb::SBDebugger::StateAsCString(state));
     }
   } else {
-    if (dap.log)
-      *dap.log << "error: SendThreadStoppedEvent() invalid process"
-               << std::endl;
+    LLDB_LOG(GetLog(DAPLog::Protocol),
+             "error: SendThreadStoppedEvent() invalid process");
```

Comment:
Do those two messages really fit into the "protocol" log category? I thought they would rather be "internal assertion failures" / "programming bugs inside lldb-dap"?

Reply:
I wasn't sure of a better name, but they do represent a kind of internal error where we're not in the expected state. This helper is meant to send the 'stopped' event and if the process isn't stopped, then there is something wrong with how we're implementing the protocol. But I'm open to suggestions on a better name.

Reply:
Not sure, I guess having this under protocol is fine after all.

Maybe we should add something like "This should never happen. Please file a bug report." to the message, though to make it clear that this should actually be unreachable?